src/translator/batching_pool.cpp

```diff
@@ -59,5 +59,11 @@ size_t BatchingPool::enqueueRequest(Ptr<Request> request) {
   return toBeFreshlyTranslated;
 }
 
+void BatchingPool::clear() {
+  for (size_t length = 0; length < bucket_.size(); length++) {
+    bucket_[length].clear();
```

Comment:
What happens if clear is called while someone wants to consume from the bucket? (I don't know the code here, just asking if it is guaranteed to not happen?)

Reply:
BatchingPool is wrapped by ThreadsafeBatchingPool, which has a lock to make sure generateBatch() and clear() are not called concurrently.